test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout.lua

```diff
@@ -0,0 +1,619 @@
+---------------------------------------------------------------------------------------------------
+-- Common module
+---------------------------------------------------------------------------------------------------
+--[[ General configuration parameters ]]
+config.defaultProtocolVersion = 2
+config.checkAllValidations = true
+config.application1.registerAppInterfaceParams.appHMIType = { "REMOTE_CONTROL" }
+config.application2.registerAppInterfaceParams.appHMIType = { "REMOTE_CONTROL" }
+
+--[[ Required Shared libraries ]]
+local json = require("modules/json")
+local utils = require('user_modules/utils')
+local actions = require("user_modules/sequences/actions")
+local commonRC = require('test_scripts/RC/commonRC')
+local runner = require('user_modules/script_runner')
+
+--[[ Tests Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Common Variables ]]
+local c = {}
+c.start = actions.start
+c.registerAppWOPTU = actions.registerAppWOPTU
+c.activateApp = actions.activateApp
+c.postconditions = actions.postconditions
+c.getMobileSession = actions.mobile.getSession
+c.getHMIConnection = actions.hmi.getConnection
+c.getHMIAppId = actions.app.getHMIId
+c.defineRAMode = commonRC.defineRAMode
+c.rpcAllowed = commonRC.rpcAllowed
+c.getAppEventName = commonRC.getAppEventName
+c.getAppRequestParams = commonRC.getAppRequestParams
+c.getHMIEventName = commonRC.getHMIEventName
+c.getHMIRequestParams = commonRC.getHMIRequestParams
+c.getHMIResponseParams = commonRC.getHMIResponseParams
+
+c.notificationTime = 0
+c.jsonFileToTable = utils.jsonFileToTable
+c.tableToJsonFile = utils.tableToJsonFile
+c.cloneTable = utils.cloneTable
+c.Step = runner.Step
+c.Title = runner.Title
+
+c.modules = { "RADIO", "CLIMATE" }
+
+c.rpcs = {}
+
+c.rpcsArray = {
```

Comment:
Technically all RPCs have default or custom timeouts, so `onResetTimeout` can be used by HMI for any RPC. Agreed that some RPCs like `Show` don't take long for SDL_HMI or GENERIC_HMI long enough to need resetting timeout, but for sake of API we should cover any and all RPCs which HMI can use `onResetTimeout` for an automated test setup. Can you please add all the RPCs in the list.

Reply:
@atiwari9
Yes, that's right current implementation in Core allows to handle `OnResetTimeout` for all requests.
Initially we created ATF tests only for those which require an input from user on HMI side and resetting of timeout is vital for them.
There are more than 30 other RPCs which don't require user input, though could be additionally covered.
Taking into account very tight time constraints it's almost impossible to create tests for all of them.
We would appreciate for the list of most important ones which can be covered. 
From our point of view we can suggest the following: 
 - `CreateInteractionChoiceSet`
 - `DeleteInteractionChoiceSet`
 - `AddCommand`
 - `AlertManeuver`
 - `ChangeRegistration`
 - `DeleteSubMenu`
 - `SetGlobalProperties`

Reply:
I assumed the request and responses are auto generated from API. But i guess we do need to define some default values here. Ok then, we can limit the RPCs to the ones implemented and additionally what you mentioned above.